### PR TITLE
Allow to force upgrade or upgrade from local ref of git

### PR DESCRIPTION
**Pull request by kiennq**
This should be merged after #162 
- Allow to force upgrade by using prefix in `quelpa-upgrade-all` or `quelpa-upgrade`. The default is to signal error whenever there's local change left. With force option, always upgrade to latest ref and signal no error.
- Allow to upgrade package from current local ref. This is done by `C-u C-u M-x quelpa-upgrade`.

### CODE REVIEW

**Comment by steckerhalter** [comment]
what's the state on this PR, can it be merged? what has to be done?

**Comment by kiennq** [comment]
> what's the state on this PR, can it be merged? what has to be done?

This should be merged after #162

File: quelpa.el [hunk 1]
```diff
@@ -1759,11 +1767,12 @@ So here we replace that with `insert-file-contents' for non-tar files."
...
     (package-install-file file)))
 
-(defun quelpa-package-install (arg)
+(defun quelpa-package-install (arg &rest plist)
   "Build and install package from ARG (a recipe or package name).
+PLIST is a plist that may modify the build and/or fetch process.
 If the package has dependencies recursively call this function to install them."
   (let* ((rcp (quelpa-arg-rcp arg))
-         (file (and rcp (quelpa-build rcp))))
+         (file (and rcp (quelpa-build (append rcp plist)))))
```
Comment (alphapapa) [inline quelpa.el]: Nitpick: Please use `when` instead of `and` to make the conditional more clear.

Comment (kiennq) [reply quelpa.el]: Done

File: quelpa.el [hunk 2]
```diff
@@ -267,7 +267,7 @@ Return nil if the package is already installed and should not be upgraded."
   "Build a package from the given recipe RCP.
 Uses the `package-build' library to get the source code and build
 an elpa compatible package in `quelpa-build-dir' storing it in
-`quelpa-packages-dir'. Return the path to the created file or nil
+`quelpa-packages-dir'.  Return the path to the created file or nil
```
Comment (steckerhalter) [inline quelpa.el]: I'm curious, why does it have two space here now? Is that an elisp rule or something?

Comment (alphapapa) [reply quelpa.el]: Yes, that is standard Emacs rules, which `checkdoc` warns about.  The simplest reason is that it allows `M-a`/`M-e` to move between sentences, which is very convenient when editing prose.

Comment (steckerhalter) [reply quelpa.el]: :+1:

**Pull request merged**